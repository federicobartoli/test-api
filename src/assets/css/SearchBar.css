.searchbar {
  max-width: var(--max-width);
  margin: 0 auto;
}

.searchbar p {
  color: var(--grey);
  text-align: center;
  padding: var(--base-spacing) 0;
}

.searchbar .filter-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: calc(2 * var(--base-spacing));
  margin-bottom: calc(4 * var(--base-spacing));
}

.searchbar .filter-container button {
  padding: var(--base-spacing);
  min-width: 300px;
  text-align: center;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 90px;
  border: 0;
  background: var(--green);
  font-size: var(--small);
  cursor: pointer;
  color: black;
}

.searchbar .filter-container button:hover {
  background: #6aea5f;
}

.searchbar .filter-container .filter-container__choices {
  display: flex;
  justify-content: center;
  gap: calc(4 * var(--base-spacing));
}

.searchbar .filter-container .filter-container__choices .filter__choice input {
  margin-right: var(--base-spacing);
}

.react-tags {
  position: relative;
  border: 1px solid var(--light-grey);
  border-radius: 1px;
  padding: var(--base-spacing);
  margin: 0 calc(2 * var(--base-spacing)) calc(2 * var(--base-spacing));
  /* clicking anywhere will focus the input */
  cursor: text;
  font-size: 16px;
  line-height: 16px;
}

.react-tags.is-focused {
  border-color: #b1b1b1;
}

.react-tags__selected {
  display: inline;
}

.react-tags__selected-tag {
  display: inline-block;
  box-sizing: border-box;
  margin: 0 6px 6px 0;
  padding: 6px 8px;
  border: 1px solid #d1d1d1;
  border-radius: 2px;
  background: #f1f1f1;

  /* match the font styles */
  font-size: inherit;
  line-height: inherit;
}

.react-tags__selected-tag:after {
  content: '\2715';
  color: #aaa;
  margin-left: 8px;
}

.react-tags__selected-tag:hover,
.react-tags__selected-tag:focus {
  border-color: #b1b1b1;
}

.react-tags__search {
  display: inline-block;

  /* match tag layout */
  padding: 7px 2px;

  /* prevent autoresize overflowing the container */
  max-width: 100%;
}

@media screen and (min-width: 30em) {
  .react-tags__search {
    /* this will become the offsetParent for suggestions */
    position: relative;
  }
}

.react-tags__search-input {
  /* prevent autoresize overflowing the container */
  max-width: 100%;

  /* remove styles and layout from this element */
  margin: 0;
  padding: 0;
  border: 0;
  outline: none;

  /* match the font styles */
  font-size: inherit;
  line-height: inherit;
  width: auto !important;
}

.react-tags__search-input::-ms-clear {
  display: none;
}

.react-tags__suggestions {
  position: absolute;
  top: 100%;
  left: 0;
  width: 100%;
}

@media screen and (min-width: 30em) {
  .react-tags__suggestions {
    width: 240px;
  }
}

.react-tags__suggestions ul {
  margin: 4px -1px;
  padding: 0;
  list-style: none;
  background: white;
  border: 1px solid #d1d1d1;
  border-radius: 2px;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.2);
}

.react-tags__suggestions li {
  border-bottom: 1px solid #ddd;
  padding: 6px 8px;
}

.react-tags__suggestions li mark {
  text-decoration: underline;
  background: none;
  font-weight: 600;
}

.react-tags__suggestions li:hover {
  cursor: pointer;
  background: #eee;
}

.react-tags__suggestions li.is-active {
  background: #b7cfe0;
}

.react-tags__suggestions li.is-disabled {
  opacity: 0.5;
  cursor: auto;
}
