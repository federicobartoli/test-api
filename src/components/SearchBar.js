import { useState, useCallback, useRef } from 'react';
import ReactTags from 'react-tag-autocomplete';
import '../assets/css/SearchBar.css';

const SearchBar = ({ handleFilter, andOr, setAndOr }) => {
  const [keyWords, setKeyWords] = useState([]);
  const reactTags = useRef();

  const onDelete = useCallback(
    (keyWordIndex) => {
      setKeyWords(keyWords.filter((_, i) => i !== keyWordIndex));
    },
    [keyWords],
  );

  const onAddition = useCallback(
    (newKeyWord) => {
      setKeyWords([...keyWords, newKeyWord]);
    },
    [keyWords],
  );

  return (
    <div className="searchbar">
      <p>
        <small>Insert a comma after each keyword</small>
      </p>
      <ReactTags
        placeholderText="Add keywords"
        allowNew={true}
        delimiters={[',']}
        ref={reactTags}
        tags={keyWords}
        onDelete={onDelete}
        onAddition={onAddition}
      />
      <div className="filter-container">
        <button onClick={() => handleFilter(keyWords)}>FILTER</button>
        <button
          onClick={() => {
            handleFilter([]);
            setKeyWords([]);
          }}
        >
          RESET
        </button>
        <div className="filter-container__choices">
          <div className="filter__choice">
            <input
              checked={!andOr}
              type="radio"
              id="or"
              name="filter"
              value="or"
              onChange={() => setAndOr(false)}
            />
            <label htmlFor="or">||</label>
          </div>

          <div className="filter__choice">
            <input
              checked={andOr}
              type="radio"
              id="and"
              name="filter"
              value="and"
              onChange={() => setAndOr(true)}
            />
            <label htmlFor="and">{`&&`}</label>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SearchBar;
