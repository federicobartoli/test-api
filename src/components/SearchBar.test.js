import React from 'react';
import SearchBar from './SearchBar';
import ReactTags from 'react-tag-autocomplete';

import { render, fireEvent, screen } from '@testing-library/react';

describe('SearchBar component', () => {
  test('UI SearchBar', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });

  test('should contain heading text', () => {
    render(<SearchBar />);
    expect(screen.getByText('Insert a comma after each keyword')).toBeVisible();
  });

  test('should call handleFilter function when filter button is pressed', () => {
    const handleFilter = jest.fn();
    render(<SearchBar handleFilter={handleFilter} />);
    const filterButton = screen.getByText('FILTER');
    fireEvent.click(filterButton);
    expect(handleFilter).toHaveBeenCalledTimes(1);
  });
});
