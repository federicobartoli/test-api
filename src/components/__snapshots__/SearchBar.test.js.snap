// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component UI SearchBar  1`] = `
<div>
  <div
    class="searchbar"
  >
    <p>
      <small>
        Insert a comma after each keyword
      </small>
    </p>
    <div
      class="react-tags"
    >
      <div
        aria-live="polite"
        aria-relevant="additions removals"
        class="react-tags__selected"
      />
      <div
        class="react-tags__search"
      >
        <div
          class="react-tags__search-wrapper"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-label="Add keywords"
            aria-owns="ReactTags"
            class="react-tags__search-input"
            placeholder="Add keywords"
            role="combobox"
            style="width: 2px;"
            value=""
          />
          <div
            style="position: absolute; width: 0px; height: 0px; visibility: hidden; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
          >
            Add keywords
          </div>
        </div>
      </div>
    </div>
    <div
      class="filter-container"
    >
      <button>
        FILTER
      </button>
      <button>
        RESET
      </button>
      <div
        class="filter-container__choices"
      >
        <div
          class="filter__choice"
        >
          <input
            checked=""
            id="or"
            name="filter"
            type="radio"
            value="or"
          />
          <label
            for="or"
          >
            ||
          </label>
        </div>
        <div
          class="filter__choice"
        >
          <input
            id="and"
            name="filter"
            type="radio"
            value="and"
          />
          <label
            for="and"
          >
            &&
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchBar component UI SearchBar 1`] = `
<div>
  <div
    class="searchbar"
  >
    <p>
      <small>
        Insert a comma after each keyword
      </small>
    </p>
    <div
      class="react-tags"
    >
      <div
        aria-live="polite"
        aria-relevant="additions removals"
        class="react-tags__selected"
      />
      <div
        class="react-tags__search"
      >
        <div
          class="react-tags__search-wrapper"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-label="Add keywords"
            aria-owns="ReactTags"
            class="react-tags__search-input"
            placeholder="Add keywords"
            role="combobox"
            style="width: 2px;"
            value=""
          />
          <div
            style="position: absolute; width: 0px; height: 0px; visibility: hidden; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
          >
            Add keywords
          </div>
        </div>
      </div>
    </div>
    <div
      class="filter-container"
    >
      <button>
        FILTER
      </button>
      <button>
        RESET
      </button>
      <div
        class="filter-container__choices"
      >
        <div
          class="filter__choice"
        >
          <input
            checked=""
            id="or"
            name="filter"
            type="radio"
            value="or"
          />
          <label
            for="or"
          >
            ||
          </label>
        </div>
        <div
          class="filter__choice"
        >
          <input
            id="and"
            name="filter"
            type="radio"
            value="and"
          />
          <label
            for="and"
          >
            &&
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
